lighthouse-core/computed/resource-summary.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('./computed-artifact.js');
+const NetworkRecords = require('./network-records.js');
+const MainResource = require('./main-resource.js');
+const URL = require('../lib/url-shim');
+const Budget = require('../config/budget.js');
+
+class ResourceSummary {
+  /**
+   * @param {LH.Artifacts.NetworkRequest} record
+   * @return {LH.Budget.ResourceType}
+   */
+  static determineResourceType(record) {
+    /** @type {Record<string, LH.Budget.ResourceType>} */
+    const mapping = {
+      'Stylesheet': 'stylesheet',
+      'Image': 'image',
+      'Media': 'media',
+      'Font': 'font',
+      'Script': 'script',
+      'Document': 'document'
+    }
+    return record.resourceType ? mapping[record.resourceType] || 'other' : 'other';
+  }
+
+  /**
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {string} mainResourceURL
+   * @return {Object<string,{resourceType: string, count: number, size: number}>}
+   */
+  static summarize(networkRecords, mainResourceURL) {
+    /** @type {Object<string,{resourceType: string, count: number, size: number}>} */
```

Comment:
I think we want the `Record` syntax on this and the above.

also looks like these are indexed by `LH.Budget.ResourceType` not just `string`?